src/libraries/System.Reflection.MetadataLoadContext/tests/src/SampleMetadata/SampleMetadata.cs

```diff
@@ -526,6 +526,30 @@ public class PropertyHolder1<T>
 
     public class DerivedFromPropertyHolder1<T> : PropertyHolder1<T> { }
 
+    public class PropertyWithNonPublicAccessors
+    {
+        // Property with public setter but private getter - should be considered public
+        public string? PublicSetterPrivateGetter { private get; set; }
+        
+        // Property with both public getter and setter - should be considered public  
+        public string? PublicGetterPublicSetter { get; set; }
+        
+        // Property with only public setter (no getter) - should be considered public
+        public string? OnlyPublicSetter { set { } }
+        
+        // Property with both private getter and setter - should not be considered public
+        private string? PrivateGetterPrivateSetter { get; set; }
+        
+        // Property with both internal getter and setter - should not be considered public
+        internal string? InternalGetterInternalSetter { get; set; }
+        
+        // Property with public getter but internal setter - should be considered public
+        public string? PublicGetterInternalSetter { get; internal set; }
+        
+        // Property with internal getter but public setter - should be considered public  
+        public string? InternalGetterPublicSetter { internal get; set; }
```

Comment:
```suggestion
        // Property with internal getter but public init - should be considered public  
        public string? PublicInitInternalGetter { init; internal get; }
```
There are more types of accessors, like `init` accessors. We should have tests for those.

I do not think that the fix is going to work correctly for these. The logic needs to mirror `ComposedOfAllPrivateMethods` logic used in the implementation of the core reflection. Otherwise, it won't match core reflection.

Reply:
I considered `init` but I thought it'd be covered since reflection exposes `init` accessors as `set`.  Adding the test is fine.  I suspect it will pass.

FWIW here's what SRM exposes:
https://github.com/dotnet/runtime/blob/4252c8d09b2ec537928f34dad269f02f167c8ce5/src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeSystem/HandleCollections.TypeSystem.cs#L1618

But we don't actually persist those through MLC reflection types:
https://github.com/dotnet/runtime/blob/664ac25dcc16cb1531f90765295ececf5b45a0e3/src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Properties/Ecma/EcmaProperty.cs#L98-L99
https://github.com/dotnet/runtime/blob/664ac25dcc16cb1531f90765295ececf5b45a0e3/src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Properties/RoProperty.cs#L109-L128

I suppose we could plumb through the `Others` but I'm not sure if anything in C# uses those yet.  I can do some research.

Reply:
I see how this is different than CoreCLR reflection, but right now I'm not seeing a small enough change without side effects to bring them closer.  Considering all accessors would introduce allocation and also would not gain anything without further changes to MLC - it's also not fixing any reported issue.

I do see that we could realized longer term benefit by sharing more of the managed code in CoreCLR reflection and MLC - but right now they are built on different metadata models - so fixing this is not in the scope of this PR.  I would be interesting in how we could do that longer term though.